sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,528 @@
+
+# Specification
+
+The proposed changes are listed as follows:
+
+Specify JSON-RPC 2.0 compatible methods and payloads for wallet interaction.
+These can be used via a browser object (i.e., via the `window.btc.request` method) or similar interfaces like WalletConnect.
+
+# Backwards Compatibility
+
+The implementation of this proposal is not necessarily backward compatible.
+However, wallets implementing the new standard are advised to maintain the previous system to support legacy applications during a transition period.
+Existing applications using the current Auth system should continue to operate, but immediate changes are recommended once this SIP is ratified.
+The Connect library should go live with the new standard, once most major wallets have adopted the new standard.
+
+# Implementation
+
+## Notes on Serialization
+
+These methods are based on JSON-RPC 2.0 for serialization.
+To adhere to a generic serializability, the following notes are given.
+Enums are serialized as human-readable strings.
+BigInts are serialized as numbers, strings, or anything that can be parsed by the JavaScript BigInt constructor.
+Bytes are serialized as hex-encoded strings (without a 0x prefix).
+Predefined formats from previous SIPs are used where applicable.
+Addresses are serialized as Stacks c32-encoded strings.
+Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+## Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered recommendations for the wallet.
+The user/wallet may choose to ignore/override them.
+Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in more generic settings and other more Ethereum inspired domains.
+In other cases (e.g. WalletConnect), the namespace may already be given by meta-data (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+#### Transaction method general params
+
+The following definitions can be used in the transaction methods.
+
+`params`
+
+- `address?`: `string` address, Stacks c32-encoded, defaults to wallets current address
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` BigInt constructor compatible value
+- `nonce?`: `number | string` BigInt constructor compatible value
+- `attachment?`: `string` hex-encoded
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+- `postConditionMode?`: `'allow' | 'deny'`
+- `sponsored?`: `boolean`, defaults to `false`
+- ~~`appDetails`~~ _removed_
+- ~~`onFinish`~~ _removed_
+- ~~`onCancel`~~ _removed_
+
+`where`
+
+- `PostCondition`: `string | object` hex-encoded or JSON representation
+
+---
+
+### Method `stx_transferStx`
```

Comment:
The structure here seems a little ill-suited. have you considered a table vs markdown lists?
the text for all of the methods seems fine, i just think the data can be presented to the reader in an easier to consume format.

Reply:
looking it over again, maybe a table is also ill-suited. 
perhaps some better formatting though?
was thinking about this, ex:
```
### stx_transferSip10Ft

`params`

- `recipient`: `string` address, Stacks c32-encoded
- `asset`: `string` address, Stacks c32-encoded, with contract name suffix
- `amount`: `number | string` BigInt constructor compatible value

`result`

- `txid`: `string` hex-encoded
- `transaction`: `string` hex-encoded raw transaction
```

Reply:
Yeah, table is also tricky. I'll try alternate headings and change how the `where` adds an additional unneeded section.